javatests/arcs/core/delegates/DelegateTest.kt

```diff
@@ -0,0 +1,39 @@
+package arcs.core.delegates
+
+import arcs.BooleanDelegate
+import arcs.NumDelegate
+import arcs.TextDelegate
+import com.google.common.truth.Truth.assertThat
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+/** Tests for [StringDecoder]. */
+@Suppress("UNCHECKED_CAST", "UNUSED_VARIABLE")
+@RunWith(JUnit4::class)
+class DelegateTest {
+
+    @Test
+    fun TestNumDelegate() {
+        var n: Double by NumDelegate()
+        assertThat(n).isEqualTo(0.0)
+        n = 10.0
+        assertThat(n).isEqualTo(10.0)
```

Comment:
TIL this is possible!  I thought property delegates were only legal on class instance variables, not locals to a method.